com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/CollectionBuilder.java

```diff
@@ -132,16 +126,10 @@ public Task<Void> create(IResource input) throws IOException, CompileExceptionEr
         List<Task<?>> embedTasks = new ArrayList<>();
         for (long hash : uniqueResources.keySet()) {
             IResource genResource = uniqueResources.get(hash);
-            taskBuilder.addOutput(genResource);
-            Task<?> embedTask = project.createTask(genResource);
-            if (embedTask == null) {
-                throw new CompileExceptionError(input,
-                                                0,
-                                                String.format("Failed to create build task for component '%s'", genResource.getPath()));
-            }
+            Task<?> embedTask = createSubTask(genResource, taskBuilder);
```

Comment:
We create files on disc for embeded objects manually here, so we have to create tasks for them either